package/endpoint_adapter/src/endpoint_adapter.c

```diff
@@ -698,287 +767,256 @@ static ssize_t handle_write_all_via_framer(handle_t *handle,
   return buffer_index;
 }
 
-static void io_loop_pubsub(handle_t *read_handle, handle_t *write_handle)
+static void io_loop_pubsub(pk_loop_t *loop, handle_t *read_handle, handle_t *write_handle)
 {
-  debug_printf("io loop begin\n");
+  if (read_handle->pk_ept != NULL) {
```

Comment:
Inverting to
```C
  if (read_handle->pk_ept == NULL) {
    PK_METRICS_UPDATE(MR, MI.rx_read_count);
  } else {
    PK_METRICS_UPDATE(MR, MI.tx_read_count);
  }
```
would save one logical hop for us poor souls with limited brain power.

Reply:
fixed see UPDATE_IO_LOOP_METRIC macro